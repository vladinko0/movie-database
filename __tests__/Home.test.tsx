import 'react-native';
import React from 'react';
import Home from '../src/screens/home';

import renderer from 'react-test-renderer';

jest.mock('react-native-really-awesome-button', () => ({
  ThemedButton: () => {},
}));
jest.mock('react-native-really-awesome-button', () => ({
  ThemedButton: () => {},
}));
const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

it('renders correctly Home screen', () => {
  const home = renderer.create(<Home />);
  expect(home).toMatchSnapshot();
});
