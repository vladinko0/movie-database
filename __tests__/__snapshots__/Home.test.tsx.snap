// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Home screen 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4283866105,
        4289848060,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "height": 50,
        }
      }
    />
  </BVLinearGradient>
</View>
`;
